src/container.cpp

```diff
@@ -265,7 +263,7 @@ ReturnValue Container::queryAdd(int32_t index, const Thing& thing, uint32_t coun
 		return RETURNVALUE_CANNOTPICKUP;
 	}
 
-	if (item == this) {
+	if (item.get() == this) {
```

Comment:
We’ve got some legacy code here, hehe!
```suggestion
	if (item == getContainer()) {
```

Reply:
Not legacy, just comparing a simpler object (naked pointer vs shared pointer)

Ideally, `queryAdd` should be a free function...